### PR TITLE
Feature/make 3x and butter permissionless

### PR DESCRIPTION
**Pull request by Amirjab21**
We still need to decide where the cron-job will live. Are we going to add a server to our Nextjs app? What architecture preferences do we have? I can make a custom server at the root of the /app folder like this example:
https://github.com/vercel/next.js/tree/canary/examples/custom-server-express

Also, something with the DAO_ROLE needs to actually run the hardhat task that changes the approval required for these selected incentives. I've provided two example bash scripts so this can be done very quickly.

When we've decided with the server architecture, I can just create an express server and schedule cron-jobs using an npm package like https://www.npmjs.com/package/cron

### CODE REVIEW

File: packages/hardhat/lib/tasks/keeper-incentive/toggleApproval.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { utils } from "ethers";
+import { ethers } from "hardhat";
+import { task } from "hardhat/config";
+import { Address } from "packages/utils/src/types";
+
+interface Args {
+  contract: string;
+  keeperincentiveaddress: Address;
```
Comment (RedVeil) [inline packages/hardhat/lib/tasks/keeper-incentive/toggleApproval.ts:8]:
```suggestion
  keeperincentiveAddress: Address;
```

File: packages/hardhat/lib/tasks/keeper-incentive/toggleApproval.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  index: string;
+}
+
+export default task("keeper-incentive:toggleApproval", "updates keeper incentive")
+  .addParam("contract", "name of contract")
+  .addParam("keeperincentiveaddress", "address of keeper incentive")
```
Comment (RedVeil) [inline packages/hardhat/lib/tasks/keeper-incentive/toggleApproval.ts:14]:
```suggestion
  .addParam("keeperIncentiveAddress", "address of keeper incentive")
```

File: packages/hardhat/lib/tasks/keeper-incentive/toggleApproval.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  .addParam("index", "index of incentive")
+  .setAction(async (args: Args, hre) => {
+    const signer = hre.ethers.provider.getSigner();
+    const keeperIncentive = await hre.ethers.getContractAt("KeeperIncentive", args.keeperincentiveaddress, signer);
```
Comment (RedVeil) [inline packages/hardhat/lib/tasks/keeper-incentive/toggleApproval.ts:18]:
```suggestion
    const keeperIncentive = await hre.ethers.getContractAt("KeeperIncentive", args.keeperIncentiveAddress, signer);
```

File: packages/hardhat/lib/utils/getNamedAccounts.ts [hunk 4]
```diff
@@ -89,6 +89,12 @@ export default function getNamedAccounts() {
       rinkeby: "0xB4AdA607B9d6b2c9Ee07A275e9616B84AC560139",
       hardhat: "0xB4AdA607B9d6b2c9Ee07A275e9616B84AC560139",
     },
+    yMim: {
```
Comment (RedVeil) [inline packages/hardhat/lib/utils/getNamedAccounts.ts:92]: the Mim references can get deleted again

File: packages/hardhat/lib/utils/getNamedAccounts.ts [hunk 5]
```diff
@@ -147,6 +153,11 @@ export default function getNamedAccounts() {
       rinkeby: "0xd632f22692FaC7611d2AA1C0D552930D43CAEd3B",
       hardhat: "0xd632f22692FaC7611d2AA1C0D552930D43CAEd3B",
     },
+    crvMimMetapool: {
```
Comment (RedVeil) [inline packages/hardhat/lib/utils/getNamedAccounts.ts:156]: same here

**Comment by RedVeil** [comment]
I think we should make the getContract function a utility, aswell as the batchMint/batchRedeem. This would allow us to only change the butter/threeX contract in the task. Both tasks (butter/threeX) would than only be differantiated by the task name and contract address.

**Comment by Amirjab21** [comment]
> I think we should make the getContract function a utility, aswell as the batchMint/batchRedeem. This would allow us to only change the butter/threeX contract in the task. Both tasks (butter/threeX) would than only be differantiated by the task name and contract address.

done - check it outttt

**Review by anthonymartin — APPROVED** [review]

**Review by RedVeil — APPROVED** [review]

**Pull request merged**